File: 66. Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
         //iterate from end of vector to beginning, since least significant value is stored at end of vector
         for (int i = digits.size() - 1; i >= 0; i--){ 
           
            if (digits[i] + 1 != 10){ //if adding 1 to vector won't create carry, then add 1
                digits[i] += 1;
                return digits;
            }

            digits[i] = 0; //set value to 0 for first part of number 10

            if (i == 0){
                digits.insert(digits.begin(), 1); //insert 1 at very beginning of vector
                return digits;
            }
            
         }

        return digits;
    }
};

File: 66. Plus One.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
      #couldn't get for i in reverse(digits) to work properly
      #translated cpp solution over to python3
        for i in range(len(digits)-1,-1,-1):
            if (digits[i]) + 1 != 10: #if value is less than 9, then add one to it and return
                digits[i] += 1
                return digits
            digits[i] = 0 #otherwise, set value to 0
            if (i == 0): #if at very beginning of value, insert 1 in front of list
                digits.insert(0,1)
                return digits
        return digits      
